### PR TITLE
Add DTLS support to net-app API

### PR DESCRIPTION
**Pull request by jukkar**
This set adds both DTLS server and client support to net-app API. Both echo-client and echo-server sample applications are modified to support DTLS.
The last two patches add "net app" shell command that gives more information about currently active net-app connections.

### CODE REVIEW

File: subsys/net/lib/app/net_app.c [hunk 1]
```diff
@@ -393,27 +440,58 @@ struct net_context *_net_app_select_net_ctx(struct net_app_ctx *ctx,
 {
 #if defined(CONFIG_NET_APP_CLIENT)
 	if (ctx->app_type == NET_APP_CLIENT) {
```
Comment (lpereira) [inline subsys/net/lib/app/net_app.c]: Can this function be split into at least two, one selecting the context for servers, and other for clients? It's quite large, full of conditional compilation, and deeply nested.

File: subsys/net/lib/app/server.c [hunk 2]
```diff
@@ -252,11 +270,11 @@ static void tls_server_handler(struct net_app_ctx *ctx,
 {
 	int ret;
 
-	NET_DBG("Starting TLS server thread for %p", ctx);
+	NET_DBG("Starting SSL server thread for %p", ctx);
```
Comment (lpereira) [inline subsys/net/lib/app/server.c]: Better to just use TLS here.  SSL is dead.

**Comment by lpereira** [comment]
I'll do a more thorough review on this one.

**Comment by jukkar** [comment]
Update according to @lpereira comments.

**Comment by jukkar** [comment]
Update because of fixing merge conflict.

**Review by pfalcon — APPROVED** [review]
Eyeballed.

**Review by nashif — APPROVED** [review]

**Pull request merged**